src/login/logind-dbus.c

```diff
@@ -2038,6 +2044,94 @@ static usec_t nologin_timeout_usec(usec_t elapse) {
         return LESS_BY(elapse, 5 * USEC_PER_MINUTE);
 }
 
+int manager_load_scheduled_shutdown(Manager *m) {
+        _cleanup_fclose_ FILE *f = NULL;
+        _cleanup_free_ char *usec = NULL,
+               *warn_wall = NULL,
+               *mode = NULL,
+               *wall_message = NULL,
+               *uid = NULL,
+               *tty = NULL;
+        _cleanup_free_ char *unescaped = NULL;
+        int r;
+
+        assert(m);
+
+        r = parse_env_file(f, SHUTDOWN_SCHEDULE_FILE,
+                        "USEC", &usec,
+                        "WARN_WALL", &warn_wall,
+                        "MODE", &mode,
+                        "WALL_MESSAGE", &wall_message,
+                        "UID", &uid,
+                        "TTY", &tty);
+        if (r == -ENOENT)
+                return 0;
+        if (r < 0) {
+                log_debug_errno(r, "Failed to parse " SHUTDOWN_SCHEDULE_FILE ": %m");
+                goto fail;
+        }
+
+        r = deserialize_usec(strnull(usec), &m->scheduled_shutdown_timeout);
+        if (r < 0)
+                goto fail;
+
+        HandleAction handle = handle_action_from_string(mode);
+        if (handle == _HANDLE_ACTION_INVALID) {
```

Comment:
we usually check for validity of enums with a simple `handle < 0`. i.e. the thinking here is that evertyhing < 0 is invalid/bad/failure, and everything >= 0 is valid/good/success